### ISSUE TITLE
Docs site should default to the latest release

### ISSUE THREAD

**Issue by poblahblahblah** [external-secrets/external-secrets#1123]
**Describe the solution you'd like**

tl;dr - the docs site should default to the latest release OR it should always include the version drop down.

Today I was working on getting SecretStores to work with JWT Auth and Vault. I was primarily looking at the docs from https://external-secrets.io/provider-hashicorp-vault/ assuming that these were the latest docs available. I couldn't find any info on using the TokenRequestAPI (https://github.com/external-secrets/external-secrets/issues/137) so I ended up digging through Issues and the code to see what I needed to do. It wasn't until I forked the repo so I could work on a docs MR that I noticed that most of the info I needed was already in there.

Confused, I finally stumbled upon https://external-secrets.io/v0.5.2/provider-hashicorp-vault/ which had the docs from the 0.5.2 release.

I think https://external-secrets.io/provider-hashicorp-vault/  should always point to master or the latest branch and _should_ include the version drop down selector.

**What is the added value?**
This would probably cut down on confusion

**Give us examples of the outcome**

N/A

**Observations (Constraints, Context, etc):**

Here are some screenshots to illustrate my point.

Here is the docs site w/o specifying a version
![Screen Shot 2022-05-09 at 4 59 38 PM](https://user-images.githubusercontent.com/51938/167517229-8ff56d8b-2ea2-4cda-bb15-8c0f84f56ac6.png)

Here is the docs site with specifying a version
![Screen Shot 2022-05-09 at 4 59 53 PM](https://user-images.githubusercontent.com/51938/167517255-7a925875-9e4b-4e68-9b82-93ca933f20fb.png)

Does this make sense?

**Comment by gusfcarvalho** [comment]
Hi @poblahblahblah ! This is being addressed at #1088

**Comment by gusfcarvalho** [comment]
This issue is now fixed, and https://external-secrets.io/provider-hashicorp-vault/ now points to a Not Found page. This unversioned url was pointing to the oldest unversioned docs we had (IIRC 0.3.6).
